Survey/Sources/Presentation/Modules/Login/LoginViewModel.swift

```diff
@@ -42,20 +77,28 @@ extension LoginViewModel {
 
     final class Input: ObservableObject {
 
-        @Published var username = ""
+        @Published var email = ""
         @Published var password = ""
         let logInTrigger: Driver<Void>
+        let emailTrigger: Driver<Void>
```

Comment:
I need emailTrigger and passwordTrigger, since i expect validation triggers only when textfield contains text at first. If i don't use emailTrigger and passwordTrigger, it starts validating and return error when Login screen appears.
In `LoginView`:
```swift
TextField("", text: $input.email)
    .modifier(PlaceholderModifier(showPlaceHolder: input.email.isEmpty, placeholder: "Email"))
    .padding(.bottom, 3.0)
    .onChange(of: input.email, perform: { newValue in
        if !newValue.isEmpty {
            emailTrigger.send(())
        }
    })
```
So when user edits textfield with at least one letter, it triggers emailTrigger to send a value (`Void`) to subscriber in LoginViewModel.
```swift
let isEmailValid = Publishers.CombineLatest(input.$email, input.emailTrigger)
    .map { $0.0.isEmailValid && !$0.0.isEmpty }
```
`isEmailValid` starts validate the string.

Does this answer your concern or i misunderstand something? 😄

Reply:
**Update**: I found `.dropFirst()` in Combine, that's similar to `.skip()` in RxSwift, work well with my approach and your suggestion as well, thank anh 😄. Fixed at https://github.com/nkhanh44/ic-survey-ios/pull/15/commits/82fc9e096095f61e8660989eb4c122fa91fd16b1